lib/protocol/V2.js

```diff
@@ -137,7 +153,12 @@ function __encrypt(data, key, footer, nonce, cb) {
 
     const header = utils.local(self);
 
-    [ data, footer, nonce ] = (utils.parse('utf-8'))(data, footer, nonce);
+    if (this._binary) {
+      [ data ] = (utils.parse('utf-8'))(data);
```

Comment:
Should this not be "binary" rather than "utf-8"?

Reply:
Good catch, I have no idea how the test was passing.